core/src/main/java/io/questdb/ServerMain.java

```diff
@@ -388,17 +387,30 @@ protected void configureWorkerPools(final WorkerPool sharedPoolQuery, final Work
 
         engine.buildMatViewGraph();
 
-        if (matViewEnabled && !isReadOnly && config.getMatViewRefreshPoolConfiguration().isEnabled()) {
-            // create dedicated worker pool for materialized view refresh
-            WorkerPool matViewRefreshWorkerPool = workerPoolManager.getInstanceWrite(
-                    config.getMatViewRefreshPoolConfiguration(),
-                    WorkerPoolManager.Requester.MAT_VIEW_REFRESH
-            );
-            setupMatViewJobs(matViewRefreshWorkerPool, engine, workerPoolManager.getSharedQueryWorkerCount());
+        if (matViewEnabled && !isReadOnly) {
+            if (config.getMatViewRefreshPoolConfiguration().getWorkerCount() > 0) {
+                // This starts mat view refresh jobs only when there is a dedicated pool for mat view refresh
+                // this will not use shared pool write because getWorkerCount() > 0
+                WorkerPool mvRefreshWorkerPool = workerPoolManager.getSharedPoolWrite(
+                        config.getMatViewRefreshPoolConfiguration(),
+                        WorkerPoolManager.Requester.MAT_VIEW_REFRESH
+                );
+
+                setupMatViewJobs(
+                        mvRefreshWorkerPool,
+                        engine,
+                        workerPoolManager.getSharedQueryWorkerCount()
+                );
+            } else {
+                log.advisory().$("mat view refresh is disabled; set ")
```

Comment:
nit: we could log one level up, i.e. when `matViewEnabled` or `!isReadOnly` is `false`. It's more likely that someone disabled mat view via the enabled switch rather than the worker count.